stream_alert_cli/cli.py

```diff
@@ -161,6 +166,27 @@ def _continue_prompt():
             return True
         return False
 
+    def _refresh_tf_state(self):
+        logging.info('Refreshing Remote State config')
+        region = self.config['region']
+        bucket = '{}.streamalert.terraform.state'.format(self.config['prefix'])
+        s3_key = self.config['tfstate_s3_key']
+        kms_key_id = 'alias/{}'.format(self.config['kms_key_alias'])
+
+        remote_state_opts = [
+            'terraform',
+            'remote',
+            'config',
+            '-backend=s3',
+            '-backend-config="bucket={}"'.format(bucket),
+            '-backend-config="key={}"'.format(s3_key),
+            '-backend-config="region={}"'.format(region),
+            '-backend-config="kms_key_id={}"'.format(kms_key_id),
+            '-backend-config="encrypt=true"'
+        ]
+
+        self.run_command([' '.join(remote_state_opts)], quiet=True, shell=True)
```

Comment:
Using `shell=True` is a potential security risk. A future downstream application with restricted permissions which only allows users to configure bucket names would then be vulnerable to shell injection. Let's try to avoid `shell=True` if possible.

I think the reason `suprocess.check_call` is not working here is because of the double-quotes in the argument names. Try different alternatives, like single quotes, no quotes, etc. See if there is a Python API that will sanitize/invoke terraform on your behalf.

If all else fails and `shell=True` is the only option, put a comment here about why we have to do it, and then a warning in the global README that tells the user that the config input is assumed to be trusted.